src/test/suite/participant/participant.test.ts

```diff
@@ -2176,6 +2176,142 @@ Schema:
       );
     });
 
+    suite('with askForNameSpace', function () {
+      const userMessages = [
+        'find all docs by a name example',
+        'what other queries can be used as an example',
+      ];
+
+      const chatRequestMock = {
+        prompt: 'localhost',
+        command: 'query',
+      };
+
+      beforeEach(function () {
+        chatContextStub = {
+          history: [
+            Object.assign(Object.create(vscode.ChatRequestTurn.prototype), {
```

Comment:
nit: I think I probably added this to some other tests a while back. Maybe we can make a helper function like `createVSCodeChatRequestTurn` that these can use. Same with `ChatResponseTurn`. To make us avoid re-writing some of the boilerplate each time.